fiftyone/utils/labels.py

```diff
@@ -1198,3 +1206,212 @@ def _get_filepath(sample_or_frame, sample):
         return sample_or_frame.filepath
     except:
         return sample.filepath
+
+
+def detections_3d_to_cuboids_2d(
+    dataset: SampleCollection,
+    camera_slice_name: str,
+    lidar_slice_name: str,
+    in_field: str,
+    out_field: str,
+    transformations: Dict[str, fou3d.TransformationType],
+    camera_params: Dict[str, Dict[str, Any]],
+    forward_transform_flags: Dict[str, List[bool]] = None,
+    batch_size: int = 1000,
+    camera_model: Callable = fou3d.pinhole_projector,
+    progress: bool = None,
+):
+    """High-level orchestration of 3D → 2D label conversion. Processes the `in_field`
+    on the `lidar_slice_name` slice of a grouped dataset, uses the transformations
+    and camera intrinsics to converts the labels to 2D polylines and saves them in
+    the `out_field` on the `camera_slice_name` slice.
+
+    Args:
+        dataset: a
+            :class:`fiftyone.core.collections.SampleCollection`
+        camera_slice_name: the name of the camera slice in the dataset
+        lidar_slice_name: the name of the LiDAR slice in the dataset
+        in_field: the name of the :class:`fiftyone.core.labels.Detection` field
+            containing 3D labels to convert
+        out_field: the name of the :class:`fiftyone.core.labels.Polylines`
+            field to populate with 2D cuboids
+        transformations: a dict mapping sample IDs to transformation tuples
+            (translation, rotation) for each sample in the dataset.
+            Translation is a 3-element list or np.ndarray, and rotation is a
+            (3, 3) list of lists or np.ndarray representing a rotation matrix.
+        camera_params: a dict mapping sample IDs to camera parameters dict which
+            contains the keys required by the camera_model for each sample in
+            the dataset.
+        forward_transform_flags: a dict mapping sample IDs to lists of booleans
+            indicating whether to apply forward or inverse transformations for
+            each transform in the list of transformations for each sample
+        batch_size (1000): number of samples to process in each batch
+        camera_model (fou3d.pinhole_projector): a callable that takes 3D points
+            in the camera coordinate system and the camera parameters dict and
+            returns the projected 2D points in pixel coordinates
+        progress (True): whether to show a progress bar during processing
+    """
+    fov.validate_grouped_non_dynamic_collection(dataset)
+    fov.validate_collection_label_fields(dataset, in_field, fol.Detections)
+
+    lidar_slice = dataset.select_group_slices(lidar_slice_name)
+    camera_slice = dataset.select_group_slices(camera_slice_name)
+    camera_slice.compute_metadata()
```

Comment:
nit: it'd be nice to pass `progress` to `compute_metadata()` as well since it also provides control over progress printing:

```py
camera_slice.compute_metadata(progress=progress)
```